internal/server/auth/method/http.go

```diff
@@ -150,13 +155,20 @@ func (m Middleware) Handler(next http.Handler) http.Handler {
 	})
 }
 
-func parts(path string) (provider, method string, ok bool) {
-	const prefix = "/auth/v1/method/oidc/"
-	if !strings.HasPrefix(path, prefix) {
-		return "", "", false
+func parts(path string) (provider, method, prefix string, ok bool) {
+	var oidcPrefix = "/auth/v1/method/oidc/"
+	if strings.HasPrefix(path, oidcPrefix) {
+		b, a, f := strings.Cut(path[len(oidcPrefix):], "/")
+		return b, a, oidcPrefix, f
+	}
+
+	var githubPrefix = "/auth/v1/method/github"
+	if strings.HasPrefix(path, githubPrefix) {
+		b, a, f := strings.Cut(path[len(githubPrefix):], "/")
+		return b, a, githubPrefix, f
 	}
 
-	return strings.Cut(path[len(prefix):], "/")
+	return "", "", "", false
```

Comment:
Pedant suggestion here, but could we make it return `(prefix, provider, method string, ok bool)`.
Just to preserve the original order of the parts.

Also, maybe pick some more meaningful variables names than `b`, `a` and `f`. I am unsure what those refer to.

You could even:
```go
const (
    oidcPrefix = "/auth/v1/method/oidc/"
    githubPrefix = "/auth/v1/method/github/"
)

func parts(path string) (prefix, provider, method string, ok bool) {
    if strings.HasPrefix(path, oidcPrefix) {
        prefix = oidcPrefix
        provider, method, ok = strings.Cut(path[len(oidcPrefix):], "/")
    } else if strings.HasPrefix(path, githubPrefix) {
        prefix = githubPrefix
        provider, method, ok = strings.Cut(path[len(githubPrefix):], "/")
    }
    
    return
}
```